models/maespa/R/met2model.MAESPA.R

```diff
@@ -174,6 +174,9 @@ met2model.MAESPA <- function(in.path, in.prefix, outfolder, start_date, end_date
     lonunits <- "W"
   }
 
+  # need to load stringr in memory for Maeswrap
+  require(stringr)
```

Comment:
@robkooper if ```stringr``` is listed as a Depends in Maeswrap, why do we need to ```require``` it here and in model2netcdf? If maeswrap is calling it's libraries incorrectly, I'd prefer to PR a fix to them than to revert our code to using ```library``` calls, which we've moved away from

Reply:
This is a nice description of the problem, following is quoted from https://stackoverflow.com/a/8638902

There is one unfortunately common exception to the advice above: if your package relies on a package A which itself "Depends" on another package B, your package will likely need to attach  A with a "Depends directive.

Reply:
So options are:
1. use the stringr as done here
2. add Maeswrap as a depends in PEcAn.MAESPA
3. work with maintainer of Maeswrap to make it so he does not have depends, but all imports.

I picked the stringr since that is the smallest change and only adds stringr to namespace. Adding the depend line seems to be overkill and adds more packages to the namespace.

Let me know which you prefer.

Reply:
So it appears that model2netcdf.MAESPA requires the Maeswrap package in order to function, and write.configs (our only truly required function for wrapping a model) requires model2netcdf, which makes Maeswrap required. Therefore Maeswrap should not be in the ```Suggests``` part of the package DESCRIPTION but in the ```Imports``` (which would be preferable to the ```Depends```).  That said, adding Maeswrap to the Imports will cause the PEcAn build to fail unless (A) Maeswrap is installed or could be installed automatically (which is not what we want UNTIL Maeswrap is updated to move rgl to Suggests) or (B) PEcAn.MAESPA is itself not required in the build. Neither are good options. @tonygardella or @robkooper have either of you been talking to @RemkoDuursma or @mdekauwe about a PR to Maeswrap moving rgl to Suggests -- that's the best solution for PEcAn, but I haven't heard whether it'd be acceptable on their side.

@robkooper it sounds like you also think Maeswrap should likewise use ```Imports``` over ```Depends```, so I guess I'm coming down in favor of your option (3) of working with Maeswrap maintainers as our Plan A.  Seems like you could test the Imports/Depends shift in your docker/DESCRIPTION.maeswrap the same way you tested moving rgl to Suggests.

Reply:
Pull request created: https://bitbucket.org/remkoduursma/maeswrap/pull-requests/1/use-imports-suggests-instead-of-depends/diff

Reply:
Used my branch of maeswrap and not changes to PEcAn and it ran successfully.